site/content/contribute/server/plugins.md

```diff
@@ -0,0 +1,42 @@
+---
+title: "Plugins"
+date: 2017-08-20T11:35:32-04:00
+weight: 5
+subsection: Server
+---
+
+# Plugins
+
+Mattermost supports plugins that offer powerful features for extending and deeply integrating with both the server and web/desktop apps.
+
+This document covers the plugin infrastructure and how to contribute to it.
+
+## Building Plugins
+
+Looking to build a plugin? [Then you want the plugin author documenation.](/extend/plugins)
+
+## Overview
+
+Plugins are generally made of at least two parts: a manifest and a server binary and/or a JavaScript bundle.
+
+The manifest tells Mattermost what the plugin is and provides a set of metadata used by the server to install and run the plugin. For more information you can see the [manifest reference](/extend/plugins/manifest-reference). Manifests may be defined in JSON or YAML.
+
+The server binary is a compiled (generally Go) program that extends the [MattermostPlugin](https://godoc.org/github.com/mattermost/mattermost-server/plugin#MattermostPlugin) struct of the [plugin](https://godoc.org/github.com/mattermost/mattermost-server/plugin) package. When enabled, the plugin's server binary is started as a process by the Mattermost server. Plugin builders then have access to interact with the Mattermost over RPC through the plugin [API](https://godoc.org/github.com/mattermost/mattermost-server/plugin#API) and [Hooks](https://godoc.org/github.com/mattermost/mattermost-server/plugin#Hooks). More information is available on the [server-side of plugins in the author documentation](/extend/plugins/server).
+
+The JavaScript bundle is a webpack-built collection of JavaScript code that will be run on the Mattermost web/desktop apps. When a plugin is enabled, the client is notified and it makes a request to add the JS bundle to the document. The plugin's client code then registers itself and components with the Mattermost client through the client's [plugin registry](https://github.com/mattermost/mattermost-webapp/blob/master/plugins/registry.js). The registry contains many methods for registering different components and callbacks. These are all stored as part of the app's [plugin reducer](https://github.com/mattermost/mattermost-webapp/blob/master/reducers/plugins/index.js). The [Pluggable](https://github.com/mattermost/mattermost-webapp/tree/master/plugins/pluggable) component is then inserted into various places in the app, allowing plugins to insert components into these locations in the UI. In some special cases, the Pluggable component is not used and we instead implement the plugs manually. More information is available on the [web app side of plugins in the author documenation](/extend/plugins/webapp).
+
+In the future there will be another component of plugins for the mobile apps, likely a React Native bundle.
```

Comment:
Maybe add "for mobile". Some people interested might not know that our mobile app is React Native.

Reply:
It does say "for the mobile apps" right before, is that not good enough?

Reply:
It is clear to me.

One thing that I’m thinking is that we should add somewhere that if you are making a plugin to change the format For a custom post type the mobile apps won’t render them as expected but instead will render the raw message formatted with our standard formatter?